lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1287 @@
+
+        self.module = module
+        self.results = {
+            'changed': False,
+            'actions': [],
+            'container': {},
+        }
+        self.name = self.module.params['name']
+        self.executable = self.module.get_bin_path(self.module.params['executable'], required=True)
+        self.image = self.module.params['image']
+        image_actions = ensure_image_exists(self.module, self.image)
+        self.results['actions'] += image_actions
+        self.state = self.module.params['state']
+        self.restart = self.module.params['force_restart']
+        self.recreate = self.module.params['recreate']
+        self.container = PodmanContainer(self.module, self.name)
+
+    def update_container_result(self, changed=True):
+        facts = self.container.get_info()
+        self.results.update({'changed': changed, 'container': facts,
+                             'ansible_facts': {'podman_container': facts}})
+        self.module.exit_json(**self.results)
+
+    def started(self):
+        if self.container.running and (self.container.different or self.recreate):
+            self.container.recreate()
+            self.results['actions'].append('recreated %s' % self.container.name)
+            self.update_container_result()
+        elif self.container.running and not self.container.different:
+            if self.restart:
+                self.container.restart()
+                self.results['actions'].append('restarted %s' % self.container.name)
+                self.update_container_result()
+            self.module.exit_json(**self.results)
+        elif not self.container.exists:
+            self.container.run()
+            self.results['actions'].append('started %s' % self.container.name)
+            self.update_container_result()
+        elif self.container.stopped and self.container.different:
+            self.container.recreate()
+            self.results['actions'].append('recreated %s' % self.container.name)
+            self.update_container_result()
+        elif self.container.stopped and not self.container.different:
+            self.container.start()
+            self.results['actions'].append('started %s' % self.container.name)
+            self.update_container_result()
+
+    def stopped(self):
+        if not self.container.exists and not self.image:
+            self.module.fail_json(msg='Cannot create container when image is not specified!')
+        if not self.container.exists:
+            self.container.create()
+            self.results['actions'].append('created %s' % self.container.name)
+            self.update_container_result()
+        if self.container.stopped:
+            self.update_container_result(changed=False)
+        elif self.container.running:
+            self.container.stop()
+            self.results['actions'].append('stopped %s' % self.container.name)
+            self.update_container_result()
+
+    def absent(self):
```

Comment:
It's usually better to use verbs for method names because they represent some actions, not states.

Reply:
@webknjaz I learnt from docker_container module: https://github.com/sshnaidm/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_container.py#L2299
I'm happy to change this and improve the current code, but in such big file there is always a place for improvement, and this process can long forever actually. Maybe we can limit some scope of work for the first iteration? It will be much easier psychologically to solve all issues at once, than not to know if they will finish at all and this is very frustrating and discouraging feeling. It just seems like it goes nowhere.
Thanks for understanding.

Reply:
ok